base_local_planner/include/base_local_planner/trajectory_cost_function.h

```diff
@@ -58,7 +58,7 @@ class TrajectoryCostFunction {
    * General updating of context values if required.
    * Subclasses may overwrite. Return a non-zero error code in case there is any error.
    */
-  virtual ExePathOutcome prepare() = 0;
+  virtual ExePathOutcome prepare(const geometry_msgs::PoseStamped& current_pose) = 0;
```

Comment:
```suggestion
  virtual ExePathOutcome prepare([[maybe_unused]] const geometry_msgs::PoseStamped& current_pose) = 0;
```

Reply:
Hmm I thought we keep this all old style c++ code to minimize changes made compared to upstream; `[[maybe_unused]]` is c++ 17 and we aren't setting `set(CMAKE_CXX_STANDARD 17)` anywhere either?

Reply:
ah ok, got it